src/3rdparty/qtokenizer/qtokenizer.h

```diff
@@ -0,0 +1,264 @@
+       to skip the next character.
+
+       \param quotes Characters that delimit quotes.
+     */
+    void setQuoteCharacters(const T& quotes) { d->quotes = quotes; }
+
+
+    /*!
+       Whether or not to return delimiters ase tokens
+       \see setQuoteCharacters
+     */
+    void setReturnQuoteCharacters(bool enable) { d->returnQuotes = enable; }
+
+
+    /*!
+       Retrieve next token.
+
+       Returns true if there are more tokens, false otherwise.
+
+       \sa next()
+     */
+    bool hasNext()
+    {
+        typename QTokenizerPrivate<T, const_iterator>::State state;
+        d->isDelim = false;
+        for (;;) {
+            d->tokenBegin = d->tokenEnd;
+            if (d->tokenEnd == d->end)
+                return false;
+            d->tokenEnd++;
+            if (d->nextChar(&state, *d->tokenBegin))
+                break;
+            if (d->returnDelimiters) {
+                d->isDelim = true;
+                return true;
+            }
+        }
+        while (d->tokenEnd != d->end && d->nextChar(&state, *d->tokenEnd)) {
+            d->tokenEnd++;
+        }
+        return true;
+    }
+
+    /*!
+       Resets the tokenizer to the starting position.
+     */
+    void reset() {
+        d->tokenEnd = d->begin;
+    }
+
+    /*!
+       Returns true if the current token is a delimiter,
+       if one more more delimiting characters have been set.
+     */
+    bool isDelimiter() const { return d->isDelim; }
+
+    /*!
+       Returns the current token.
+
+       Use \c hasNext() to fetch the next token.
+     */
+    T next() const {
+        int len = d->tokenEnd-d->tokenBegin;
+        const_iterator tmpStart = d->tokenBegin;
+        if (!d->returnQuotes && len > 1 && d->isQuote(*d->tokenBegin)) {
+            tmpStart++;
+            len -= 2;
+        }
+        return T(tmpStart, len);
+    }
+
+private:
+    friend class QStringTokenizer;
+    QSharedPointer<QTokenizerPrivate<T, const_iterator> > d;
+};
+
+class QStringTokenizer : public QTokenizer<QString, QString::const_iterator> {
+public:
+    QStringTokenizer(const QString &string, const QString &delim) :
+        QTokenizer<QString, QString::const_iterator>(string, delim) {}
+    /**
+     * @brief Like \see next(), but returns a lightweight string reference
+     * @return A reference to the token within the string
+     */
+    QStringRef stringRef() {
+        int begin = d->tokenBegin-d->begin;
+        int end = d->tokenEnd-d->tokenBegin;
+        if (!d->returnQuotes && d->isQuote(*d->tokenBegin)) {
+            begin++;
+            end -= 2;
+        }
+        return QStringRef(&d->string, begin, end);
+    }
+};
+
+typedef QTokenizer<QByteArray, QByteArray::const_iterator> QByteArrayTokenizer;
+typedef QTokenizer<std::string, std::string::const_iterator> StringTokenizer;
+typedef QTokenizer<std::wstring, std::wstring::const_iterator> WStringTokenizer;
+
+QT_END_NAMESPACE
+
+#endif // TOKENIZER_H
+
```

Comment:
No special marking of member variables (we use the prepended '_') and implementations in the header file - both not our style, but I assume that's cool in Qt?